### PR TITLE
Add utilities for copying files.

### LINKED ISSUE #353: Transferring files to pod
Hello,
I tried to send files to a pod with the lib using the Exec object. It works, but it can't ensure the integrity of the file.
Here is an example of how I do it : 

```
public class ExecExample {
    public static void main(String[] args) throws IOException, ApiException {
        String podName = "nginx-4217019353-k5sn9";
        String namespace = "default";

        ApiClient client = Config.defaultClient();
        Configuration.setDefaultApiClient(client);

        Exec exec = new Exec();
        final Process proc =
                exec.exec(
                        namespace,
                        podName,
                        new String[]{"tee" ,"/tmp/test"},
                        true,
                        false);

        try {
            FileInputStream file = new FileInputStream("C:/test");
            OutputStream pod = proc.getOutputStream();
            IOUtils.copy(file, pod);
            pod.flush();
            IOUtils.closeQuietly(file);
            IOUtils.closeQuietly(pod);
        } catch (IOException ex) {
            ex.printStackTrace();
        }

        proc.destroy();

        System.exit(proc.exitValue());
    }
}
```

The problem is that there are missing bytes in my pod file. Not always the same amount and it's depending of the original file size. 
For example : 
- Sent 1024 bytes -> always received 1024 bytes = OK
- Sent 2048 bytes -> always received 2047 bytes = KO
- Sent 4096 bytes -> always received 4095 bytes = KO
- Sent 8192 bytes -> always received 8190 bytes = KO

I tested with the command `tar xf - -C /tmp` and sending a tar file, and I also tested the commands with kubectl. No problems in the both cases.
It's maybe a buffer problem, but I don't know when. IOUtils.copy() return the right amount of bytes.

Development environment : Windows
Kubernetes Java library version : 2.0.0
Kubernetes version : 1.8.2

### PR DESCRIPTION
**Pull request by brendandburns**
Closes #353

### CODE REVIEW

**Comment by kondapally1989** [comment]
@brendandburns 
looks good to me.
may be we can add todo statements for
1. is tar available.
2. and can function names be like copyFileFrom/copyDirectoryFrom so that in future we can also add some thing like copyFileTo/copyDirectoryTo  ?

adding  copyFileTo  Pod  may not  be appropriate as it defeats the purpose of docker / kubernetes. but may be for testing people can use it.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!

File: util/src/main/java/io/kubernetes/client/Copy.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+          }
+        }
+      }
+    }
+  }
+
+  public static void copyFile(String namespace, String pod, String srcPath, Path dest)
+      throws ApiException, IOException {
+    Copy c = new Copy();
+    InputStream is = c.copyFile(namespace, pod, null, srcPath);
```
Comment (kondapally1989) [inline util/src/main/java/io/kubernetes/client/Copy.java]: i think travis build is throwing error here , string to Path

File: util/src/main/java/io/kubernetes/client/Copy.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+   *
+   * @param apiClient The api client to use.
+   */
+  public Copy(ApiClient apiClient) {
+    super(apiClient);
+  }
+
+  public InputStream copyFileFromPod(String namespace, String pod, String srcPath)
+      throws ApiException, IOException {
+    return copyFile(namespace, pod, null, srcPath);
```
Comment (kondapally1989) [inline util/src/main/java/io/kubernetes/client/Copy.java]: is it copyFileFromPod ?

File: util/src/main/java/io/kubernetes/client/Copy.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  }
+
+  public InputStream copyFileFromPod(V1Pod pod, String srcPath) throws ApiException, IOException {
+    return copyFile(pod, null, srcPath);
```
Comment (kondapally1989) [inline util/src/main/java/io/kubernetes/client/Copy.java]: is it copyFileFromPod ?

File: util/src/main/java/io/kubernetes/client/Copy.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  }
+
+  public InputStream copyFileFromPod(V1Pod pod, String container, String srcPath)
+      throws ApiException, IOException {
+    return copyFile(
```
Comment (kondapally1989) [inline util/src/main/java/io/kubernetes/client/Copy.java]: is it copyFileFromPod ?

File: util/src/main/java/io/kubernetes/client/Copy.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  public void copyFileFromPod(String namespace, String name, String srcPath, Path destination)
+      throws ApiException, IOException {
+    try (InputStream is = copyFileFromPod(namespace, name, srcPath);
+         FileOutputStream fos = new FileOutputStream(destination.toFile())) {
+      ByteStreams.copy(is, fos);
+      fos.flush();
+    }
+  }
+
+  public void copyDirectoryFromPod(V1Pod pod, String srcPath, Path destination)
```
Comment (kondapally1989) [inline util/src/main/java/io/kubernetes/client/Copy.java]: is it copyDirectoryToPod ?

File: util/src/main/java/io/kubernetes/client/Copy.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+  public void copyDirectoryFromPod(V1Pod pod, String container, String srcPath, Path destination)
+      throws ApiException, IOException {
+    copyDirectoryToPod(pod.getMetadata().getNamespace(), pod.getMetadata().getName(), container, srcPath, destiation);
+  }
+
+
+  public void copyDirectoryFromPod(String namespace, String pod, String srcPath, Path destination)
+      throws ApiException, IOException {
+    copyDirectoryToPod(namespace, pod, null, srcPath, destiation);
```
Comment (kondapally1989) [inline util/src/main/java/io/kubernetes/client/Copy.java]: destination

File: util/src/main/java/io/kubernetes/client/Copy.java (hunk 1 above)
Comment (kondapally1989) [inline util/src/main/java/io/kubernetes/client/Copy.java]: copyFileFromPod?

**Comment by kondapally1989** [comment]
/hold cancel
i did /lgtm in one of previous comment but deleted the comment as ci build broke and added hold.

@brendandburns  once you fix those compile issues, i think it will automatically merge

**Comment by brendandburns** [comment]
@kondapally1989 updated, thanks for the patience!

**Comment by kondapally1989** [comment]
@brendandburns  format issue in compile time.

**Comment by brendandburns** [comment]
@kondapally1989 sorry for the noise, I swear I fixed the formatting, but apparently not.  Please take another look.

**Pull request merged**